### PR TITLE
Check code formatting

### PR DESCRIPTION
**Pull request by i-ky**
This is a proof of concept for code formatting check by Travis CI using clang-format tool. It's definitely *not* ready to merge into master right now, but I think that agreement on coding guidelines and some tool to check it automatically is what we need ASAP to start moving forward at a greater pace.

@v-zhuravlev , feel free to adjust .clang-format file to your liking. Meaning of options is described [here](https://clang.llvm.org/docs/ClangFormatStyleOptions.html).

Perhaps, worth making formatting check an independent job/stage/whatever in Travis CI to separate it from build and tests.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -16,6 +16,8 @@ before_install:
 script:
   - ./autogen.sh 
   - ./configure --enable-zabbix-${ZBX_HEADERS_VERSION}
+  - clang-format -i --style=file src/*.c
+  - git diff --exit-code
```
Comment (v-zhuravlev) [inline .travis.yml]: Any idea why `make` is broken after clang-format applied?
https://travis-ci.org/v-zhuravlev/libzbxmodbus/jobs/426184921#L2224

Comment (i-ky) [reply .travis.yml]: Because of [this](https://travis-ci.org/v-zhuravlev/libzbxmodbus/jobs/426184921#L832-L834), I suppose.

Comment (v-zhuravlev) [reply .travis.yml]: Thank you, fixed

File: .clang-format [hunk 2]
```diff
@@ -61,7 +61,7 @@ PenaltyExcessCharacter: 1000000
 PenaltyReturnTypeOnItsOwnLine: 100
 PointerAlignment: Right
 ReflowComments: true
-SortIncludes: true
+SortIncludes: false # sysinc.h must be before module.h
```
Comment (i-ky) [inline .clang-format]: Just a note. Sections of code can be [protected](https://clang.llvm.org/docs/ClangFormatStyleOptions.html#disabling-formatting-on-a-piece-of-code) from auto formatting like so:
```
/* clang-format off */
#include "sysinc.h"
#include "module.h"
/* clang-format on */
```

Comment (v-zhuravlev) [reply .clang-format]: thanks, changed

**Comment by v-zhuravlev** [comment]
Done?

**Comment by i-ky** [comment]
Seems so :)
Thank you for your input!

**Pull request merged**